tests/ballerina-compiler-api-test/src/test/java/io/ballerina/semantic/api/test/ExpressionTypeTestNew.java

```diff
@@ -321,6 +321,15 @@ public void testStartAction() {
         assertType(101, 10, 101, 21, NIL);
     }
 
+    @Test
+    public void testFutureResultType() {
+        TypeSymbol type = getExprType(350, 31, 350, 38);
+        assertEquals(type.typeKind(), FUTURE);
+        Optional<TypeSymbol> typeParameter = ((FutureTypeSymbol) type).typeParameter();
+        assertTrue(typeParameter.isPresent());
+        assertEquals(typeParameter.get().typeKind(), INT);
+    }
+
```

Comment:
Let's add another test case to cover the type of `barFn()` within the start action. i.e., `(354, 17, 354, 24)`